landscape/client/manager/usermanager.py

```diff
@@ -118,6 +130,15 @@ def _add_user(self, message):
             message["home-number"],
         )
 
+    def _add_user_core(self, message):
+        """Run an C{add-user} operation on Core."""
+        return self._management.add_user(
+            message["username"],
+            message["email"],
+            message["sudoer"],
+            message["force-managed"],
+        )
```

Comment:
I don't love this. It's a violation of interface-dependency. I'd prefer it if we modified `add_user` to have a signature that can support both, even if that's just:

    def add_user(self, message):

That way the differences between adding users on core and non-core is contained completely within the `UserManagement` class and doesn't leak up to the `Manager`.

Changing what message types we respond to is _ok_, but alternatively you could simply have the message types we do not respond to be no-ops in `SnapdUserManagement`.